aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,285 @@
+ 
+A French resource file could contain the following:
+
+========================  ===============================  
+Key                       Value    
+========================  ===============================  
+<i>Hello</i> <b>{0}!</b>  <i>Bonjour</i> <b>{0}!</b>  
+========================  ===============================
+
+The rendered view would contain the HTML markup from the resource file.
+
+To use a shared resource file in a view, inject ``IHtmlLocalizer<T>``:
+
+.. literalinclude:: localization/sample/Views/Test/About.cshtml
+  :language: HTML
+  :linenos:
+  :emphasize-lines: 5,12
+
+DataAnnotations localization
+------------------------------------
+
+DataAnnotations error messages are localized with ``IStringLocalizer<Type>``. Using the option ``ResourcesPath = "Resources"``, the error messages in ``RegisterViewModel`` can be stored in either of the following paths:
+
+- Resources/ViewModels.Account.RegisterViewModel.fr.resx
+- Resources/ViewModels/Account/RegisterViewModel.fr.resx
+  
+.. literalinclude:: localization/sample/ViewModels/Account/RegisterViewModel.cs
+  :language: c#
+  :lines: 9-26
+  :dedent: 3
+  
+Currently, the run time doesn't look up localized strings for non-validation attributes. In the code above, the "Email" (from ``[Display(Name = "Email")]``) will not be localized.
+  
+Localization middleware
+-----------------------------------
+
+The current culture on a request is set in the localization :doc:`/fundamentals/middleware`. The localization middleware is enabled in the ``Configure`` method of *Startup.cs* file.
+
+.. literalinclude:: localization/sample/StartupConfig.cs
+  :language: c#
+  :linenos:
+  :lines: 106-107, 136-159
+  :dedent: 6
+
+``UseRequestLocalization`` initializes a ``RequestLocalizationMiddleware`` object. On every request the list of ``RequestCultureProviders`` in the ``RequestLocalizationOptions`` is enumerated and the first non-null provider is used. The default providers come from the ``RequestLocalizationOptions`` class:
+
+#. ``QueryStringRequestCultureProvider``
+#. ``CookieRequestCultureProvider``
+#. ``AcceptLanguageHeaderRequestCultureProvider``
+
+The default list goes from most specific to least specific. Later in the article I'll show how you can change the order and even add a custom localization provider. If there are no non-null providers, the ``DefaultRequestCulture`` is used.
+    
+QueryStringRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition the providing the primary mechanism some app will use to set the culture and UI culture, adding a query string to the URL is useful for debugging and testing code. Unless you change the ``RequestCultureProviders`` list, a query string to the will always win as the localization provider. You pass the query string parameters ``culture`` and ``ui-culture``. The following example sets the specific culture (language and region) to Spanish/Mexico:
```

Comment:
We should have a link that defines what the current culture and UI culture are in .NET.